### PR TITLE
BLD: fix for cython3 compatibility

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by tacaswell** [comment]
xref https://github.com/scipy/scipy/issues/17234#issuecomment-1493423202

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [build-system]
 requires = [
-    "Cython ~=0.29",
+    "Cython >=0.29.31",
```
Comment (takluyver) [inline pyproject.toml:3]: I'm a bit wary of declaring Cython 3 compatibility before the final version is out (I think the `~=0.29` specifier is equivalent to `>=0.29, <1`). Though I see they're onto betas now, so hopefully it's not far off.

Comment (tacaswell) [reply pyproject.toml:3]: maybe `>=0.29.31, <1` ?  We need to put a floor on the patch version so that the keyword exists.

Comment (takluyver) [reply pyproject.toml:3]: Yup, if we do that, I think it's fine.

Comment (tacaswell) [reply pyproject.toml]: done.

**Pull request merged**